packages/shared/src/components/MainFeedLayout.tsx

```diff
@@ -224,26 +262,41 @@ export default function MainFeedLayout({
         emptyScreen: <SearchEmptyScreen />,
       };
     }
-    if (!query.query) {
+
+    if (!config.query) {
       return null;
     }
 
     const getVariables = () => {
-      if (isUpvoted) {
-        return { ...query.variables, period: periods[selectedPeriod].value };
+      if (isUpvoted || feedWithDateRange.includes(tab)) {
+        return { ...config.variables, period: periods[selectedPeriod].value };
+      }
+
+      if (isExplore) {
+        const laptopValue =
+          tab === ExploreTabs.ByDate || isExploreLatest ? 1 : 0;
+        const finalAlgo = isLaptop ? laptopValue : selectedAlgo;
+        return {
+          ...config.variables,
+          ranking: algorithms[finalAlgo].value,
+        };
       }
 
       if (isSortableFeed) {
         return {
-          ...query.variables,
+          ...config.variables,
           ranking: algorithms[selectedAlgo].value,
         };
       }
 
-      return query.variables;
+      return config.variables;
     };
 
     const variables = getVariables();
+    const classes = isExplore && {
+      wrapper: '!min-h-0',
+      emptyScreen: 'mt-8 h-auto',
+    };
```

Comment:
Why does this one only have specific needs?
Is there no other way we can maybe fix it?

Reply:
The placement of the empty screen is not the absolute center of the screen anymore, and with the existing classes, it produces a scrollbar unnecessarily. Arguably not needed, but UX-wise, I think it is better to not have the scrollbar.

Reply:
Hmm and we can't put it on "all" empty states?

Reply:
Let me check, though as I remember it looks fine on the current setup. Just with the addition of the header tabs that broke it.

Reply:
What do you think of the screenshots below? The CTA became more centered on the screen than being more at the bottom side. Not sure on mobile if it is better to be near the bottom or the center. However, on laptop, we also removed the scrollbar.

Before:
![Screenshot 2024-06-11 at 5 16 02 PM](https://github.com/dailydotdev/apps/assets/13744167/5f46c8bf-efcf-4388-8784-97b944b79152)
![Screenshot 2024-06-11 at 5 15 56 PM](https://github.com/dailydotdev/apps/assets/13744167/69cfed51-652c-4f93-a254-e0a786d2686b)
![Screenshot 2024-06-11 at 5 15 46 PM](https://github.com/dailydotdev/apps/assets/13744167/0ea2fdd9-d15f-45fa-bcb7-a593415552fc)

After adding the same classes:
![Screenshot 2024-06-11 at 5 14 03 PM](https://github.com/dailydotdev/apps/assets/13744167/92df47e6-d6cb-4d1d-b910-69b3d8e44085)
![Screenshot 2024-06-11 at 5 14 19 PM](https://github.com/dailydotdev/apps/assets/13744167/41ab2e4f-4309-49fa-a236-0906e778c85e)
![Screenshot 2024-06-11 at 5 13 56 PM](https://github.com/dailydotdev/apps/assets/13744167/4eb21498-2204-4deb-a83e-1f8cfd9be857)

Reply:
Such a edge-case anyway I'm fine with it.